components/omega/src/drivers/standalone/OceanDriver.cpp

```diff
@@ -0,0 +1,61 @@
+//===-- drivers/standalone/OceanDriver.cpp - Standalone driver --*- C++ -*-===//
+//
+//
+//===----------------------------------------------------------------------===//
+
+#include "OceanDriver.h"
+#include "DataTypes.h"
+#include "OceanState.h"
+#include "OmegaKokkos.h"
+#include "TimeMgr.h"
+
+#include "mpi.h"
+
+#include <iostream>
+
+int main(int argc, char **argv) {
+
+   int ErrAll;
+   int Err1;
+   int Err2;
+
+   MPI_Init(&argc, &argv); // initialize MPI
+   Kokkos::initialize();   // initialize Kokkos
+
+   // Time management objects
+   OMEGA::Calendar OmegaCal;
+   OMEGA::TimeInstant CurrTime;
+   OMEGA::Alarm EndAlarm;
+
+   Err1 = OMEGA::ocnInit(MPI_COMM_WORLD, OmegaCal, CurrTime, EndAlarm);
+   if (Err1 != 0)
+      LOG_ERROR("Error initializing OMEGA");
+
+   while (Err1 == 0 && !(EndAlarm.isRinging())) {
+
+      Err1 = OMEGA::ocnRun(CurrTime, EndAlarm);
+
+      if (Err1 != 0)
+         LOG_ERROR("Error advancing Omega run interval");
+   }
+
+   Err2 = OMEGA::ocnFinalize(CurrTime);
+   if (Err2 != 0)
+      LOG_ERROR("Error finalizing OMEGA");
+
+   ErrAll = abs(Err1) + abs(Err2);
```

Comment:
This is making me wonder if enforcing positive (or negative) error codes would be a good idea.

Reply:
Agreed. I think there is consensus that we need a more consistent and coherent policy for error logging and error handling. Currently the `Config.get` functions return `-1` when a variable or group isn't found, but in most places we are using positive values for errors.

Reply:
Indeed, in the ErrorHandler design doc I'm putting together, this is listed as a requirement so that accumulated error codes don't sum to zero. So we'll take care of it when we replace these error checks.